### PR TITLE
feat: Handle successful ingestion via Lambda

### PR DESCRIPTION
**Pull request by metakuni**
This is a companion PR to https://github.com/chanzuckerberg/single-cell-infra/pull/474

### Reviewers
**Functional:** 
@ebezzi 

**Readability:** 
@danieljhegeman 

---

## Changes
- modify: replaced ingestion success handler batch task with a lambda

## QA steps (optional)
tested on rdev:
- https://us-west-2.console.aws.amazon.com/states/home?region=us-west-2#/executions/details/arn:aws:states:us-west-2:699936264352:execution:dp-rdev-rdev-kuni-sfn:ee95b381-d5cc-45c9-8987-8c3a03c996b6_1639160087

### CODE REVIEW

File: .happy/terraform/modules/sfn/main.tf [hunk 1]
```diff
@@ -151,41 +151,9 @@ resource "aws_sfn_state_machine" "state_machine" {
...
-            "ErrorEquals": [
-              "States.TaskFailed"
-            ],
-            "IntervalSeconds": 1,
-            "BackoffRate": 2,
-            "MaxAttempts": 2
-          }
-        ]
+        "InputPath": "$.dataset_uuid",
+        "Resource": "${var.lambda_success_handler}",
```
Comment (metakuni) [inline .happy/terraform/modules/sfn/main.tf:155]: Batch replaced by lambda

File: Dockerfile.handle_success [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM public.ecr.aws/lambda/python:3.8
```
Comment (metakuni) [inline Dockerfile.handle_success:1]: Docker image for the new lambda handler

File: backend/corpora/dataset_processing/process.py [hunk 3]
```diff
@@ -550,8 +550,8 @@ def main():
             from backend.corpora.dataset_processing.process_seurat import process
 
             process(dataset_id, os.environ["ARTIFACT_BUCKET"])
-        elif step_name == "handle-success":
```
Comment (metakuni) [inline backend/corpora/dataset_processing/process.py:553]: Handle success moved to lambda

File: backend/corpora/lambdas/handle_success/handle_success.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    Lambda function invoked by the ingestion step function that updates the processing status
+    for the specified dataset to SUCCESS
+    :param event: Lambda's event object
+    :param context: Lambda's context object
+    :return:
+    """
+    dataset_uuid = event["dataset_uuid"]
+
+    try:
+        update_db(dataset_uuid, processing_status=dict(processing_status=ProcessingStatus.SUCCESS))
```
Comment (metakuni) [inline backend/corpora/lambdas/handle_success/handle_success.py:20]: That's all, folks!

File: backend/corpora/lambdas/handle_success/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+psycopg2-binary>=2.8.5
```
Comment (metakuni) [inline backend/corpora/lambdas/handle_success/requirements.txt:1]: These versions have been superseded, but thought they should be kept in sync with `backend/corpora/upload_failures/requirements.txt`.

The package versions should all be updated at the same time.

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 6]
```diff
@@ -31,6 +31,7 @@ locals {
   frontend_image_repo          = local.secret["ecrs"]["frontend"]["url"]
   backend_image_repo           = local.secret["ecrs"]["backend"]["url"]
   upload_image_repo            = local.secret["ecrs"]["processing"]["url"]
+  lambda_upload_success_repo   = local.secret["ecrs"]["upload_success"]["url"]
```
Comment (metakuni) [inline .happy/terraform/modules/ecs-stack/main.tf:34]: ECR repo for the new lambda's Docker image (added in https://github.com/chanzuckerberg/single-cell-infra/pull/474/files)

File: .happy/terraform/modules/ecs-stack/main.tf [hunk 7]
```diff
@@ -181,6 +196,7 @@ module upload_sfn {
   job_queue_arn        = local.job_queue_arn
   role_arn             = local.sfn_role_arn
   custom_stack_name    = local.custom_stack_name
+  lambda_success_handler = module.upload_success_lambda.arn
```
[the remark this answers is not in the document]
Comment (metakuni) [reply .happy/terraform/modules/ecs-stack/main.tf:199]: Fixed

File: .github/workflows/push-rdev.yml [hunk 8]
```diff
@@ -23,7 +23,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        component: ['frontend', 'backend', 'upload_failures', 'processing']
+        component: ['frontend', 'backend', 'upload_failures', 'upload_success', 'processing']
```
Comment (metakuni) [inline .github/workflows/push-rdev.yml:26]: New Docker image for the ingestion success handler lambda.

**Review by ebezzi — APPROVED** [review]

**Pull request merged**